modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -405,8 +393,13 @@ public void redeploy(String artifactName, Optional<Properties> deploymentPropert
 
     deploymentTemplate.preRedeploy(artifact);
 
+    Optional<Registry> optionalRegistry = ofNullable(artifact.getRegistry());
+    Optional<ArtifactStoppedListener> optionalArtifactStoppedListener =
+        optionalRegistry.isPresent() ? optionalRegistry.get().lookupByName(ARTIFACT_STOPPED_LISTENER) : Optional.empty();
```

Comment:
no need to use the intermediate Optional just for this. a simple null check will do.